src/main/java/io/jenkins/pluginhealth/scoring/scores/DeprecatedPluginScoring.java

```diff
@@ -30,16 +30,37 @@
 import io.jenkins.pluginhealth.scoring.model.ScoreResult;
 import io.jenkins.pluginhealth.scoring.probes.DeprecatedPluginProbe;
 
+import io.jenkins.pluginhealth.scoring.model.updatecenter.UpdateCenter;
+import io.jenkins.pluginhealth.scoring.service.ProbeService;
+import io.jenkins.pluginhealth.scoring.service.UpdateCenterService;
+
 import org.springframework.stereotype.Component;
 
+import java.io.IOException;
+
 @Component
 public class DeprecatedPluginScoring extends Scoring {
     private static final float COEFFICIENT = .8f;
     private static final String KEY = "deprecation";
 
+    final ProbeContext probeContext;
+    private final ProbeService probeService;
+
     @Override
     protected ScoreResult doApply(Plugin plugin) {
         final ProbeResult deprecatedPluginProbeResult = plugin.getDetails().get(DeprecatedPluginProbe.KEY);
+
+
+        final UpdateCenter updateCenter = updateCenterService.fetchUpdateCenter();
```

Comment:
Hey @

I have added this code to for the following step:

> Then, if you provide a ProbeContext with an UpdateCenter which doesn't contain the plugin in the map of plugins, then your probe should return a FAILURE, meaning the plugin is no longer published

I was unable to figure out how to check whether the plugin exists in the map of plugin? 

Should I write a new method to return a boolean value?

Because `probeService.getProbeContext(plugin.getName(), updateCenter)` this returns `ProbeContext` type. 

And I do not know how to check it.

Reply:
Because this wasn't the place to implement the requirement.